### PR TITLE
Export Tarifverbundkarte

### PR DESCRIPTION
**Pull request by danji90**
# How to

Open tarifverbunkarte topic then open the menu "Grossformatiges PDF exportieren" 

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/menus/ExportMenu/ExportMenu.js [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  });
+  const { t } = useTranslation();
+  const ref = useRef();
+
+  useEffect(() => {
+    let timeout = null;
+    if (maxCanvasSize) {
+      return () => {};
+    }
+    timeout = setTimeout(() => {
```
Comment (friedjoff) [inline src/menus/ExportMenu/ExportMenu.js:136]: Why do we need `setTimeout`?

File: src/utils/canvasSize.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
+/* eslint-disable no-console */
```
Comment (friedjoff) [inline src/utils/canvasSize.js:1]: Can we clean up this code to reduce usage of `eslint-disable`?

File: package.json [hunk 3]
```diff
@@ -39,16 +41,18 @@
...
     "redux": "4.0.5",
     "redux-debounced": "0.5.0",
     "redux-thunk": "2.3.0",
     "style-it": "2.1.4",
-    "styled-components": "5.1.0"
+    "styled-components": "5.1.0",
+    "svgson": "^5.2.1"
   },
   "peerDependencies": {
+    "canvg": "^3.0.7",
```
Comment (friedjoff) [inline package.json:55]: Do we need this peer dependency?

File: src/components/ExportButton/ExportButton.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { jsPDF as JsPDF } from 'jspdf';
+import { useTranslation } from 'react-i18next';
+import { parse as parseSvg, stringify as stringifySvgObject } from 'svgson';
+import { makeStyles } from '@material-ui/core';
+import CanvasSaveButton from 'react-spatial/components/CanvasSaveButton';
+import Canvg from 'canvg';
+import { ReactComponent as Loader } from './loader.svg';
+import legend from './tarifverbund_legend.svg';
```
Comment (friedjoff) [inline src/components/ExportButton/ExportButton.js:11]: Can we make this a bit more generic to support other topics?

File: src/components/ExportButton/ExportButton.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+          // svgson parse string to json to access values
+          const svgJson = await parseSvg(svgString).then((json) => json);
+          const legendElements = svgJson.children.find(
+            (tag) => tag.attributes.id === 'legend_elements',
+          ).children;
+
+          // Set date DE
+          legendElements.find(
+            (element) => element.attributes.id === 'date_DE',
```
Comment (friedjoff) [inline src/components/ExportButton/ExportButton.js:114]: This looks very brittle. How about adding some unique placeholder strings in the SVG file which can be automatically replaced by values configured in `exportConfig` using some regex?

**Review by friedjoff — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (danji90) [reply package.json:55]: @friedjoff yes we do, to create the Canvas from the legend SVG in https://github.com/geops/trafimage-maps/pull/900/files#diff-c7b6ed3c8399fa412faf30a02d036cae7fab066e4d35894a410a75fc26a763ccR140

Comment (danji90) [reply package.json:55]: no we don't

File: src/components/ExportButton/ExportButton.js (hunk 4 above)
Comment (danji90) [reply src/components/ExportButton/ExportButton.js]: I moved the legend data url to the topic exportConfig, so in the future we can handle exports on topic level

File: src/components/ExportButton/ExportButton.js (hunk 5 above)
Comment (danji90) [reply src/components/ExportButton/ExportButton.js]: I ditched the svgson stuff, which was really unnecessary considering your approach of replacing unique strings.

File: src/menus/ExportMenu/ExportMenu.js (hunk 1 above)
Comment (danji90) [reply src/menus/ExportMenu/ExportMenu.js:157]: Removed

File: src/utils/canvasSize.js (hunk 2 above)
Comment (oterral) [reply src/utils/canvasSize.js:1]: Yes should clean up the code.  I was scare to break the code :D

**Pull request merged**